aws/src/index.js

```diff
@@ -349,9 +350,37 @@ class S3HttpCollector extends Collector {
 }
 
 /* eslint-disable no-param-reassign */
-const processLogText = (cloudWatchLogs, tagRegexMap) => {
+const processLogText = (cloudWatchLogs, tagRegexMap, arn) => {
+  // derive source
+  const keys = arn.split(':');
+  const region = keys[3];
+  const accountId = keys[4];
+
+  // default value
+  let source = accountId + '-' + region;
+
+  // derive source for lambda function or EC2 instance
+  const logGroup = cloudWatchLogs.logGroup;
+  const logStream = cloudWatchLogs.logStream;
+
+  if(logGroup.startsWith('/aws/lambda/')) {
+    const logGroupKeys = logGroup.split('/');
+    source = logGroupKeys[logGroupKeys.length - 1];
+  } else if (logStream.startsWith('i-')) {
+    const logStreamKeys = logGroup.split('-');
+    source = logStreamKeys[logStreamKeys.length - 1];
+  }
+
```

Comment:
Can this be part of utility method to derive source?

Reply:
Agreed. Made the necessary changes.